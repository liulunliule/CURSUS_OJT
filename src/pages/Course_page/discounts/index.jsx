import React from "react";
import "./index.scss";
import { Button, Collapse, DatePicker, Input, Select, Table } from "antd";
import { Link } from "react-router-dom";
import { DeleteOutlined, EditOutlined } from "@ant-design/icons";

function Discounts() {
    const dataSource = [
        {
            key: "1",
            no: "IT-001",
            course: "Course Title Here",
            startdate: "02 November 2019",
            enddate: "19 November 2019",

            discount: "20%",

            status: "Active",
            action: "",
        },
    ];
    const columns = [
        {
            title: "Item No.",
            dataIndex: "no",
            key: "no",
        },
        {
            title: "Course",
            dataIndex: "course",
            key: "course",
        },
        {
            title: "Start Date",
            dataIndex: "startdate",
            key: "startdate",
        },
        {
            title: "End Date",
            dataIndex: "enddate",
            key: "enddate",
        },
        {
            title: "Discount",
            dataIndex: "discount",
            key: "discount",
        },

        {
            title: "Status",
            dataIndex: "status",
            key: "status",
            render: (text) => <b style={{ color: "red" }}>{text}</b>,
        },
        {
            title: "Action",
            dataIndex: "action",
            key: "action",
            render: (text, record) => (
                <div>
                    <Button
                        icon={<EditOutlined />}
                        style={{ marginRight: 8 }}
                    />
                    <Button icon={<DeleteOutlined />} danger />
                </div>
            ),
        },
    ];

    return (
        <div className="discounts">
            <Collapse
                className="discounts_collapse"
                items={[
                    {
                        key: "1",
                        label: (
                            <h4 className="discounts_collapse_title">
                                New Discount
                            </h4>
                        ),
                        children: (
                            <div className="discounts__content">
                                <form action="" className="discount-form">
                                    <div className="discount_input">
                                        <label htmlFor="">Course*</label>
                                        <Select
                                            mode="tags"
                                            style={{
                                                width: "100%",
                                            }}
                                            placeholder="Select Course"
                                            options={[
                                                {
                                                    value: "1",
                                                    label: "Course Title Here",
                                                },
                                                {
                                                    value: "2",
                                                    label: "Course Title Here",
                                                },
                                                {
                                                    value: "3",
                                                    label: "Course Title Here",
                                                },
                                                {
                                                    value: "4",
                                                    label: "Course Title Here",
                                                },
                                                {
                                                    value: "5",
                                                    label: "Course Title Here",
                                                },
                                                {
                                                    value: "6",
                                                    label: "Course Title Here",
                                                },
                                                {
                                                    value: "7",
                                                    label: "Course Title Here",
                                                },
                                                {
                                                    value: "8",
                                                    label: "Course Title Here",
                                                },
                                            ]}
                                        />
                                    </div>
                                    <div className="discount_input">
                                        <label htmlFor="">
                                            Discount Amount
                                        </label>
                                        <Input
                                            placeholder="Percent (eg. 20 for 20%)"
                                            type="number"
                                        />
                                    </div>
                                    <div className="discount_input">
                                        <label htmlFor="">Start Date</label>
                                        <DatePicker
                                            style={{
                                                width: "100%",
                                            }}
                                        />
                                    </div>
                                    <div className="discount_input">
                                        <label htmlFor="">End Date</label>
                                        <DatePicker
                                            style={{
                                                width: "100%",
                                            }}
                                        />
                                    </div>
                                </form>

                                <button className="discount_btn">
                                    Save Changes
                                </button>
                            </div>
                        ),
                    },
                ]}
            />

            <Table
                className="discounts_table"
                dataSource={dataSource}
                columns={columns}
            />
        </div>
    );
}

export default Discounts;
