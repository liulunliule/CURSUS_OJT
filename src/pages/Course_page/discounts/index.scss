.discounts {
    margin-top: 32px;

    &_collapse {
        border-radius: 5px;
        background-color: #fff;

        &_title {
            margin-top: 5px;
        }
    }
    &__content {
        padding: 20px 31px;

        .discount-form {
            width: 70%;
            display: grid;
            grid-template-columns: repeat(2, 1fr);
            gap: 20px;

            label {
                font-size: 14px;
                font-weight: 500;
                margin-bottom: 10px;
                color: #333;
            }
        }
        .discount_btn {
            margin-top: 30px;
            height: 40px;
            border-radius: 20px;
            border: 0;
            color: #fff;
            font-weight: 500;
            background: #ed2a26;
            width: 35%;
            text-align: center;

            &:hover {
                background-color: #333;
            }
        }
    }

    &_table {
        margin-top: 31px;
    }
}
