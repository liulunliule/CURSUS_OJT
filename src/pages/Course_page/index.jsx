import React from "react";
import "./index.scss";
import { BookOutlined } from "@ant-design/icons";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import {
    faBook,
    faBullhorn,
    faDownload,
    faTag,
    faUpload,
} from "@fortawesome/free-solid-svg-icons";
import { Link, NavLink, Outlet } from "react-router-dom";
import { useSelector } from "react-redux";

function Course_page() {
    const isShowAll = useSelector((state) => state.savedCourse.isShowAll);
    return (
        <div className={`course_page ${isShowAll ? "active" : ""}`}>
            <div className="course_page_hero">
                <h2 className="course_page_hero-title">
                    <FontAwesomeIcon icon={faBook} /> Courses
                </h2>
                <div className="course_page_hero-content">
                    <h1>
                        <FontAwesomeIcon icon={faBook} size="lg" />
                        Jump Into Course Creation
                    </h1>
                    <Link to="/thirdlayout/create_new_course">
                    <button className="course_page_hero-btn">
                        Create Your Course
                    </button>
                    </Link>
                </div>
            </div>

            <div className="course_page_cta">
                <div className="course_page_cta-list">
                    <NavLink
                        to="/thirdlayout/course_page/my_courses"
                        className="course_page_cta-item"
                    >
                        <FontAwesomeIcon
                            icon={faBook}
                            className="course-page_icon"
                        />{" "}
                        My Courses
                    </NavLink>

                    <NavLink
                        to="/thirdlayout/course_page/my_purchases"
                        className="course_page_cta-item"
                    >
                        <FontAwesomeIcon
                            icon={faDownload}
                            className="course-page_icon"
                        />
                        My Purchases
                    </NavLink>

                    <NavLink
                        to="/thirdlayout/course_page/upcoming_course"
                        className="course_page_cta-item"
                    >
                        <FontAwesomeIcon
                            icon={faUpload}
                            className="course-page_icon"
                        />
                        Upcoming Courses
                    </NavLink>

                    <NavLink
                        to="/thirdlayout/course_page/discounts"
                        className="course_page_cta-item"
                    >
                        <FontAwesomeIcon
                            icon={faTag}
                            className="course-page_icon"
                        />
                        Discounts
                    </NavLink>

                    <NavLink
                        to="/thirdlayout/course_page/promotion"
                        className="course_page_cta-item"
                    >
                        <FontAwesomeIcon
                            icon={faBullhorn}
                            className="course-page_icon"
                        />
                        Promotion
                    </NavLink>
                </div>
            </div>
            <Outlet />
        </div>
    );
}

export default Course_page;
