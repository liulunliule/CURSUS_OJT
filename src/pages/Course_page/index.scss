.course_page {
    margin-top: 60px;
    margin-left: 240px;
    background-color: #f7f7f7;
    padding: 30px;
    transition: margin-left 0.5s ease;

    &.active{
        transition: margin-left 0.5s ease;
        margin-left: 0;
    }

    &_hero {
        &-title {
            display: flex;
            align-items: center;
            column-gap: 10px;
            font-size: 20px;
            font-weight: 500;
            color: #333;
        }

        &-content {
            display: flex;
            justify-content: space-between;
            align-items: center;
            margin-top: 38px;
            background-color: #fff;
            padding: 31px 21px;

            h1 {
                display: flex;
                column-gap: 10px;
                font-size: 24px;
                color: #333;
                font-weight: 400;
            }

            .course_page_hero-btn {
                font-size: 14px;
                font-weight: 500;
                color: #fff;
                background: #ed2a26;
                padding: 0px 20px;
                border-radius: 3px;
                border: 0;
                height: 40px;
            }
        }
    }

    &_cta {
        margin-top: 51px;

        &-list {
            padding-left: 0;
            display: grid;
            grid-template-columns: repeat(5, 1fr);
            align-items: center;
            background: #fff;
            border-radius: 3px;
            border: 1px solid #efefef;

            .course_page_cta-item {
                padding-left: 0;
                list-style: none;
                width: 244px;
                display: flex;
                align-items: center;
                justify-content: center;
                padding: 16.8px 14px;
                font-size: 14px;
                font-weight: 500;
                color: #333;
                border-radius: 3px;

                .course-page_icon {
                    margin-right: 8px;
                }

                &.active {
                    background-color: #ed2a26;
                    color: #fff;
                }
            }
        }
    }
}
