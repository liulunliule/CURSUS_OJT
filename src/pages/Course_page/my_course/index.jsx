import React from "react";
import "./index.scss";
import { Button, Table } from "antd";

import { Link } from "react-router-dom";
import { DeleteOutlined, EditOutlined } from "@ant-design/icons";
function My_course() {
    const dataSource = [
        {
            key: "1",
            no: "IT-001",
            title: "Course Title Here",
            date: "06 April 2020 | 08:31",
            sales: 15,
            parts: 5,
            category: "Web Development",
            status: "Active",
            action: "",
        },
        {
            key: "2",
            no: "IT-002",
            title: "Course Title Here",
            date: "05 April 2020 | 05:15",
            sales: 14,
            parts: 5,
            category: "Graphic Design",
            status: "Active",
            action: "",
        },
        {
            key: "3",
            no: "IT-003",
            title: "Course Title",
            date: "03 April 2020 | 01:30",
            sales: 12,
            parts: 6,
            category: "Bootstrap",
            status: "Active",
            action: "",
        },
    ];
    const columns = [
        {
            title: "Item No.",
            dataIndex: "no",
            key: "no",
        },
        {
            title: "Title",
            dataIndex: "title",
            key: "title",
        },
        {
            title: "Publish Date",
            dataIndex: "date",
            key: "date",
        },

        {
            title: "Sales",
            dataIndex: "sales",
            key: "sales",
        },
        {
            title: "Parts",
            dataIndex: "parts",
            key: "parts",
        },
        {
            title: "Category",
            dataIndex: "category",
            key: "category",
            render: (text) => <Link>{text}</Link>,
        },
        {
            title: "Status",
            dataIndex: "status",
            key: "status",
            render: (text) => <b style={{ color: "red" }}>{text}</b>,
        },
        {
            title: "Action",
            dataIndex: "action",
            key: "action",
            render: (text, record) => (
                <div>
                    <Button
                        icon={<EditOutlined />}
                        style={{ marginRight: 8 }}
                    />
                    <Button icon={<DeleteOutlined />} danger />
                </div>
            ),
        },
    ];
    return (
        <div className="my_course">
            <Table dataSource={dataSource} columns={columns} />
        </div>
    );
}

export default My_course;
