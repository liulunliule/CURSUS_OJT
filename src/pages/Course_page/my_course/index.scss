.my_course {
    margin-top: 32px;
}

.ant-table-wrapper .ant-table-thead > tr > th {
    background-color: #ffecec;
}


