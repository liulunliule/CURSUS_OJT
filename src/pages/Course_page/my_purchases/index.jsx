import React from "react";
import "./index.scss";
import { Button, Table } from "antd";
import {
    CloudDownloadOutlined,
    DeleteOutlined,
    EditOutlined,
    PrinterOutlined,
} from "@ant-design/icons";
import { Link } from "react-router-dom";
function My_purchases() {
    const dataSource = [
        {
            key: "1",
            no: "IT-001",
            title: "Course Title Here",
            vendor: "Zoena Singh",
            date: "25 March 2020",
            delivery: "Download",
            price: "$15",
            category: "Web Development",
            status: "Active",
            action: "",
        },
        {
            key: "2",
            no: "IT-002",
            title: "Course Title Here",
            vendor: "Rock William",
            date: "20 March 2020",
            delivery: "Download",
            price: "$20",
            category: "C++",
            status: "Active",
            action: "",
        },
    ];
    const columns = [
        {
            title: "Item No.",
            dataIndex: "no",
            key: "no",
        },
        {
            title: "Title",
            dataIndex: "title",
            key: "title",
        },
        {
            title: "Vendor",
            dataIndex: "vendor",
            key: "vendor",
            render: (text) => <Link>{text}</Link>,
        },

        {
            title: "Category",
            dataIndex: "category",
            key: "category",
            render: (text) => <Link>{text}</Link>,
        },
        {
            title: "Delivery Type",
            dataIndex: "delivery",
            key: "delivery",
            render: (text) => <b style={{ color: "red" }}>{text}</b>,
        },
        {
            title: "Price",
            dataIndex: "price",
            key: "price",
        },
        {
            title: "Purchase Date",
            dataIndex: "date",
            key: "date",
        },
        {
            title: "Action",
            dataIndex: "action",
            key: "action",
            render: (text, record) => (
                <div>
                    <Button
                        icon={<CloudDownloadOutlined />}
                        style={{ marginRight: 8 }}
                    ></Button>

                    <Button
                        icon={<DeleteOutlined />}
                        style={{ marginRight: 8 }}
                        danger
                    />
                    <Button
                        icon={<PrinterOutlined />}
                        style={{ marginRight: 8 }}
                    />
                </div>
            ),
        },
    ];
    return (
        <div className="my_purchases">
            <Table dataSource={dataSource} columns={columns} />
        </div>
    );
}

export default My_purchases;
