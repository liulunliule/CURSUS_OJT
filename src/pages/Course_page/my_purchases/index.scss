.my_purchases {
    margin-top: 32px;
}
