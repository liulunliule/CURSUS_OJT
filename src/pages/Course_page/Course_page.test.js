import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom'; // Import MemoryRouter
import { store } from '../../redux/store';
import Course_page from '.';

test('test Course page', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter>
        <Course_page/>
      </MemoryRouter>
    </Provider>
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
