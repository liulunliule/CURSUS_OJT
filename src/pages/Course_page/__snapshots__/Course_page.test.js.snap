// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Course page 1`] = `
<div
  className="course_page "
>
  <div
    className="course_page_hero"
  >
    <h2
      className="course_page_hero-title"
    >
      <svg
        aria-hidden="true"
        className="svg-inline--fa fa-book "
        data-icon="book"
        data-prefix="fas"
        focusable="false"
        role="img"
        style={Object {}}
        viewBox="0 0 448 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M96 0C43 0 0 43 0 96L0 416c0 53 43 96 96 96l288 0 32 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l0-64c17.7 0 32-14.3 32-32l0-320c0-17.7-14.3-32-32-32L384 0 96 0zm0 384l256 0 0 64L96 448c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16l192 0c8.8 0 16 7.2 16 16s-7.2 16-16 16l-192 0c-8.8 0-16-7.2-16-16zm16 48l192 0c8.8 0 16 7.2 16 16s-7.2 16-16 16l-192 0c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
          fill="currentColor"
          style={Object {}}
        />
      </svg>
       Courses
    </h2>
    <div
      className="course_page_hero-content"
    >
      <h1>
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-book fa-lg "
          data-icon="book"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 0C43 0 0 43 0 96L0 416c0 53 43 96 96 96l288 0 32 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l0-64c17.7 0 32-14.3 32-32l0-320c0-17.7-14.3-32-32-32L384 0 96 0zm0 384l256 0 0 64L96 448c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16l192 0c8.8 0 16 7.2 16 16s-7.2 16-16 16l-192 0c-8.8 0-16-7.2-16-16zm16 48l192 0c8.8 0 16 7.2 16 16s-7.2 16-16 16l-192 0c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Jump Into Course Creation
      </h1>
      <a
        href="/thirdlayout/create_new_course"
        onClick={[Function]}
      >
        <button
          className="course_page_hero-btn"
        >
          Create Your Course
        </button>
      </a>
    </div>
  </div>
  <div
    className="course_page_cta"
  >
    <div
      className="course_page_cta-list"
    >
      <a
        className="course_page_cta-item"
        href="/thirdlayout/course_page/my_courses"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-book course-page_icon"
          data-icon="book"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M96 0C43 0 0 43 0 96L0 416c0 53 43 96 96 96l288 0 32 0c17.7 0 32-14.3 32-32s-14.3-32-32-32l0-64c17.7 0 32-14.3 32-32l0-320c0-17.7-14.3-32-32-32L384 0 96 0zm0 384l256 0 0 64L96 448c-17.7 0-32-14.3-32-32s14.3-32 32-32zm32-240c0-8.8 7.2-16 16-16l192 0c8.8 0 16 7.2 16 16s-7.2 16-16 16l-192 0c-8.8 0-16-7.2-16-16zm16 48l192 0c8.8 0 16 7.2 16 16s-7.2 16-16 16l-192 0c-8.8 0-16-7.2-16-16s7.2-16 16-16z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
         
        My Courses
      </a>
      <a
        className="course_page_cta-item"
        href="/thirdlayout/course_page/my_purchases"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-download course-page_icon"
          data-icon="download"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M288 32c0-17.7-14.3-32-32-32s-32 14.3-32 32l0 242.7-73.4-73.4c-12.5-12.5-32.8-12.5-45.3 0s-12.5 32.8 0 45.3l128 128c12.5 12.5 32.8 12.5 45.3 0l128-128c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0L288 274.7 288 32zM64 352c-35.3 0-64 28.7-64 64l0 32c0 35.3 28.7 64 64 64l384 0c35.3 0 64-28.7 64-64l0-32c0-35.3-28.7-64-64-64l-101.5 0-45.3 45.3c-25 25-65.5 25-90.5 0L165.5 352 64 352zm368 56a24 24 0 1 1 0 48 24 24 0 1 1 0-48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        My Purchases
      </a>
      <a
        className="course_page_cta-item"
        href="/thirdlayout/course_page/upcoming_course"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-upload course-page_icon"
          data-icon="upload"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M288 109.3L288 352c0 17.7-14.3 32-32 32s-32-14.3-32-32l0-242.7-73.4 73.4c-12.5 12.5-32.8 12.5-45.3 0s-12.5-32.8 0-45.3l128-128c12.5-12.5 32.8-12.5 45.3 0l128 128c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L288 109.3zM64 352l128 0c0 35.3 28.7 64 64 64s64-28.7 64-64l128 0c35.3 0 64 28.7 64 64l0 32c0 35.3-28.7 64-64 64L64 512c-35.3 0-64-28.7-64-64l0-32c0-35.3 28.7-64 64-64zM432 456a24 24 0 1 0 0-48 24 24 0 1 0 0 48z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Upcoming Courses
      </a>
      <a
        className="course_page_cta-item"
        href="/thirdlayout/course_page/discounts"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-tag course-page_icon"
          data-icon="tag"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 80L0 229.5c0 17 6.7 33.3 18.7 45.3l176 176c25 25 65.5 25 90.5 0L418.7 317.3c25-25 25-65.5 0-90.5l-176-176c-12-12-28.3-18.7-45.3-18.7L48 32C21.5 32 0 53.5 0 80zm112 32a32 32 0 1 1 0 64 32 32 0 1 1 0-64z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Discounts
      </a>
      <a
        className="course_page_cta-item"
        href="/thirdlayout/course_page/promotion"
        onClick={[Function]}
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-bullhorn course-page_icon"
          data-icon="bullhorn"
          data-prefix="fas"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M480 32c0-12.9-7.8-24.6-19.8-29.6s-25.7-2.2-34.9 6.9L381.7 53c-48 48-113.1 75-181 75l-8.7 0-32 0-96 0c-35.3 0-64 28.7-64 64l0 96c0 35.3 28.7 64 64 64l0 128c0 17.7 14.3 32 32 32l64 0c17.7 0 32-14.3 32-32l0-128 8.7 0c67.9 0 133 27 181 75l43.6 43.6c9.2 9.2 22.9 11.9 34.9 6.9s19.8-16.6 19.8-29.6l0-147.6c18.6-8.8 32-32.5 32-60.4s-13.4-51.6-32-60.4L480 32zm-64 76.7L416 240l0 131.3C357.2 317.8 280.5 288 200.7 288l-8.7 0 0-96 8.7 0c79.8 0 156.5-29.8 215.3-83.3z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
        Promotion
      </a>
    </div>
  </div>
</div>
`;
